contrib/make-release.py

```diff
@@ -0,0 +1,120 @@
+import os
+import re
+import subprocess
+import sys
+
+args = None
+
+
+def exec(cmd):
+    """Execute given command"""
+    return subprocess.check_output(cmd).decode().strip()
+
+
+def parse_options():
+    """Handle command line options"""
+    ap = argparse.ArgumentParser(description='Make a new release')
+    ap.add_argument('bump',
+                    choices=['major', 'minor', 'patch'],
+                    help='which version part to bump')
+    ap.add_argument('-v',
+                    '--verbose',
+                    action='store_true',
+                    help='be more verbose')
+    global args
+    args = ap.parse_args()
+
+    logging.basicConfig(format='[%(levelname)s] %(message)s')
+    if args.verbose:
+        logging.getLogger().setLevel(level=logging.DEBUG)
+    else:
+        logging.getLogger().setLevel(level=logging.INFO)
+
+
+def identify_next_version():
+    """Figure out the new version number"""
+    try:
+        curversion = exec(['git', 'describe', '--tags', '--match', 'cvc5-*'])
+    except:
+        logging.error('git describe was unable to produce a proper version')
+        sys.exit(1)
+    logging.debug('git version info: {}'.format(curversion))
+
+    re_release = re.compile('^cvc5-(\d+)\.(\d+)\.(\d+)')
+    m = re_release.match(curversion)
+    if m:
+        major, minor, patch = map(int, m.groups())
+        if args.bump == 'major':
+            major += 1
+            minor = 0
+            patch = 0
+        elif args.bump == 'minor':
+            minor += 1
+            patch = 0
+        elif args.bump == 'patch':
+            patch += 1
+        version = "{}.{}.{}".format(major, minor, patch)
+        logging.debug('target version: {}'.format(version))
+        return version
+
+    logging.error(
+        "Did not understand current git version: '{}'".format(curversion))
+    sys.exit(1)
+
+
+def generate_cmake_version_file(version, is_release):
+    """Update the cmake version file"""
+    filename = os.path.join(os.path.dirname(os.path.dirname(__file__)),
+                            'cmake/version-base.cmake')
+    tpl = open(filename + '.template').read()
+    tpl = tpl.replace('{{VERSION}}', version)
+    tpl = tpl.replace('{{IS_RELEASE}}', 'true' if is_release else 'false')
+    open(filename, 'w').write(tpl)
+
+
+def make_release_commit(version):
+    """Make the release commit"""
+    tagname = 'cvc5-{}'.format(version)
+    exec(['git', 'add', 'cmake/version-base.cmake'])
+    exec(['git', 'commit', '-m', 'Bump version to {}'.format(version)])
+    exec(['git', 'tag', tagname])
+    return tagname
+
+
+def make_post_release_commit(version):
+    """Make the post-release commit"""
+    exec(['git', 'add', 'cmake/version-base.cmake'])
+    exec(['git', 'commit', '-m', 'Start post-release for {}'.format(version)])
+    return exec(['git', 'rev-parse', 'HEAD'])
+
+
+if __name__ == '__main__':
+    parse_options()
+
+    # Compute next version
+    version = identify_next_version()
```

Comment:
Shouldn't we bump the version in the post-release commit only?

Reply:
In the context of a release, we have three commits to consider:
- pre-release (the last commit before you run this script): `LAST_RELEASE` is whatever the last release was, and `IS_RELEASE` is false.
- release (the tagged commit created by this script): `LAST_RELEASE` is the new release, and `IS_RELEASE` is true
- post-release (the second commit from this script): `LAST_RELEASE` is still the same, `IS_RELEASE` is false

I see that `LAST_RELEASE` my be slightly misleading for the release commit itself, though.

Reply:
Ah right, I see, makes sense.

Reply:
As per offline discussion: we will have the last tag in `CVC5_LAST_RELEASE`, but have `version.cmake` print `CVC5_LAST_RELEASE + 0.0.1` when the current commit is not a tag.